query/expression.go

```diff
@@ -34,7 +34,8 @@ type EvaluationContext struct {
 	API          api.API          // Api to obtain metadata from
 	Timerange    api.Timerange    // Timerange to fetch data from
 	SampleMethod api.SampleMethod // SampleMethod to use when up/downsampling to match the requested resolution
-	Predicate    api.Predicate
+	Predicate    api.Predicate    // Predicate to apply to TagSets prior to fetching
+	FetchLimit   *int             // A limit on the number of fetches which may be performed
```

Comment:
By leaving it as an `int*`, only the code that actually cares about it (namely, metric fetching) needs to modify it. If it were an `int`, then every single recursive expression `.Evaluate()` call would need to return an extra parameter (either the entire context, or just the number of things consumed).